src/components/Search/SearchTitle.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useMemo } from 'react'
+import useResponsive from 'decentraland-gatsby/dist/hooks/useResponsive'
+import Responsive from 'semantic-ui-react/dist/commonjs/addons/Responsive'
+import { Back } from 'decentraland-ui/dist/components/Back/Back'
+import TokenList from 'decentraland-gatsby/dist/utils/dom/TokenList'
+import { Header } from 'decentraland-ui/dist/components/Header/Header'
+import { navigate } from 'gatsby-plugin-intl'
+import locations from '../../modules/locations'
+import './SearchTitle.css'
+import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+import { useSearchParams } from '../../hooks/useSearchParams'
+
+export function SearchTitle() {
+  const l = useFormatMessage()
+  const responsive = useResponsive()
+  const { search } = useSearchParams()
+  const isMobile = responsive({ maxWidth: Responsive.onlyMobile.maxWidth })
+  const isTablet = responsive({ maxWidth: Responsive.onlyTablet.maxWidth })
+  const [size, maxLength] = useMemo(() => {
+    if (isMobile) return ['medium', 15]
+    if (isTablet) return ['large', 24]
+    else return ['huge', 23]
+  }, [isMobile, isTablet])
+  const formattedSearch = useMemo(() => search && search.length > maxLength ? search.substring(0, maxLength) : search, [search, maxLength])
```

Comment:
I think we can limit this with css using ellipsis. And the `size` variable with media queries.